hack/verify-staging-meta-files.sh

```diff
@@ -36,8 +36,8 @@ exceptions=(
 )
 
 RESULT=0
-for repo in ${staging_repos[@]}; do
-  for filename in ${expected_filenames[@]}; do
+for repo in "${staging_repos[@]}"; do
```

Comment:
FWIW I saw a much less complicated looking approach for getting "the list of staging repos" in https://github.com/kubernetes/kubernetes/pull/74332, is that insufficient?

Reply:
using quotes here is wrong, since it prevents splitting (which we want)

Reply:
@ixdy 
I've tested whether quoting changes the behavior of the script. The result is NO!
First Script (without quotes):
```
#!/bin/bash

expected_filenames=(
  .github/PULL_REQUEST_TEMPLATE.md
  code-of-conduct.md
  LICENSE
  OWNERS
  README.md
  SECURITY_CONTACTS
)

for filename in ${expected_filenames[@]}; do
	echo $filename
done
=====
OUTPUT:
.github/PULL_REQUEST_TEMPLATE.md
code-of-conduct.md
LICENSE
OWNERS
README.md
SECURITY_CONTACTS
```
First Script (with quotes):
```
#!/bin/bash

expected_filenames=(
  .github/PULL_REQUEST_TEMPLATE.md
  code-of-conduct.md
  LICENSE
  OWNERS
  README.md
  SECURITY_CONTACTS
)

for filename in "${expected_filenames[@]}"; do
	echo $filename
done
OUTPUT:
.github/PULL_REQUEST_TEMPLATE.md
code-of-conduct.md
LICENSE
OWNERS
README.md
SECURITY_CONTACTS

```

Reply:
Double quotes around $@ (and similarly, ${array[@]}) prevents globbing and word splitting of individual elements, while still expanding to multiple separate arguments.

Let's say you have three arguments: baz, foo bar and *

"$@" will expand into exactly that: baz, foo bar and *

$@ will expand into multiple other arguments: baz, foo, bar, file.txt and otherfile.jpg

Since the latter is rarely expected or desired, ShellCheck warns about it.

Ref: https://github.com/koalaman/shellcheck/wiki/SC2068

Reply:
you're right, I think I'm confusing this with the case where it isn't already an array.